File: crates/ontologos-dotnet/csharp/tests/Ontologos.Tests/SmokeTest.cs
using System.Text;
using Xunit;

namespace Ontologos.Tests;

public sealed class SmokeTest
{
    [Fact]
    public void VersionMatchesRelease()
    {
        Assert.Equal("1.1.3", OntologosInfo.Version());
    }

    [Fact]
    public void BuilderClassifyEl()
    {
        using var builder = new OntologyBuilder();
        builder.AddClass("http://example.org/Pizza");
        builder.AddClass("http://example.org/Food");
        builder.SubclassOf("http://example.org/Pizza", "http://example.org/Food");
        using var ontology = builder.Build();
        using var reasoner = new Reasoner(ontology, "el");
        var report = reasoner.Classify();
        Assert.Contains("\"status\":\"classified\"", report);
        Assert.Contains("subsumption_count", report);
    }

    [Fact]
    public void FromBytesStrictFunctionalSyntax()
    {
        const string ofn = """
            Prefix(:=<http://example.org/>)
            Ontology(<http://example.org/o>
              Declaration(Class(:A))
              Declaration(Class(:B))
              SubClassOf(:A :B)
            )
            """;
        using var ontology = Ontology.FromBytes(Encoding.UTF8.GetBytes(ofn));
        Assert.True(ontology.AxiomCount >= 1L);
    }

    [Fact]
    public void SharedOntologyMutationSync()
    {
        using var builder = new OntologyBuilder();
        builder.AddClass("http://example.org/A");
        builder.AddClass("http://example.org/B");
        using var ontology = builder.Build();
        using var reasoner = new Reasoner(ontology, "el");
        Assert.Equal(0L, ontology.AxiomCount);
        reasoner.AddSubclassOf("http://example.org/A", "http://example.org/B");
        Assert.Equal(1L, ontology.AxiomCount);
    }
}

File: crates/ontologos-java/java/src/test/java/dev/ontologos/SmokeTest.java
package dev.ontologos;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

class SmokeTest {
    @Test
    void versionMatchesRelease() {
        assertEquals("1.1.4", Ontologos.version());
    }

    @Test
    void builderClassifyEl() {
        try (OntologyBuilder builder = new OntologyBuilder()) {
            builder.addClass("http://example.org/Pizza");
            builder.addClass("http://example.org/Food");
            builder.subclassOf("http://example.org/Pizza", "http://example.org/Food");
            try (Ontology ontology = builder.build();
                    Reasoner reasoner = new Reasoner(ontology, "el")) {
                String report = reasoner.classify();
                assertTrue(report.contains("\"status\":\"classified\""));
                assertTrue(report.contains("subsumption_count"));
            }
        }
    }

    @Test
    void fromBytesStrictFunctionalSyntax() {
        String ofn =
                """
                Prefix(:=<http://example.org/>)
                Ontology(<http://example.org/o>
                  Declaration(Class(:A))
                  Declaration(Class(:B))
                  SubClassOf(:A :B)
                )""";
        try (Ontology ontology = Ontology.fromBytes(ofn.getBytes(java.nio.charset.StandardCharsets.UTF_8))) {
            assertTrue(ontology.getAxiomCount() >= 1L);
        }
    }

    @Test
    void sharedOntologyMutationSync() {
        try (OntologyBuilder builder = new OntologyBuilder()) {
            builder.addClass("http://example.org/A");
            builder.addClass("http://example.org/B");
            try (Ontology ontology = builder.build();
                    Reasoner reasoner = new Reasoner(ontology, "el")) {
                assertEquals(0L, ontology.getAxiomCount());
                reasoner.addSubclassOf("http://example.org/A", "http://example.org/B");
                assertEquals(1L, ontology.getAxiomCount());
            }
        }
    }
}
